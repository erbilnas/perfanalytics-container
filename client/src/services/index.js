import axios from 'axios'

const getMeasures = async (startDate, endDate) => {
    const options = {
        "url": "http://localhost:8000/measures",
        "method": "GET",
        "params": {
            "startDate": startDate,
            "endDate": endDate,
        },
    }

    return await axios(options).then((response) => { return response.data.measures })
}

export { getMeasures }