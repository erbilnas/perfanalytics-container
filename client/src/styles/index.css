@media screen and (max-width: 528px) {
  .ant-picker-panel-container {
    width: 58% !important;
    overflow: auto !important;
    margin-left: 30px !important;
  }
}

.ant-layout {
  background-color: white !important;
}
