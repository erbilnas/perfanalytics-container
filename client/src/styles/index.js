import styled from 'styled-components'

const DatetimePickerContainer = styled.div`
    position: absolute;
    left: 50%;
    top: 10%;
    -ms-transform: translate(-50%, -50%);
    transform: translate(-50%, -50%);
    @media screen and (max-width: 528px) {
        width: 90%;
    }
`
const DatetimePickerLabel = styled.p`
    margin: 0 0 10px 10px;
`

const ChartContainer = styled.div`
    position: absolute;
    left: 50%;
    top: 15%;
    -ms-transform: translate(-50%, -50%);
    transform: translate(-50%, 0%);
`

const ChartLabel = styled.span`
    margin: 0 0 0 45%;
`

const LoaderSpan = styled.span`
    margin: 0 0 0 15px;
`

export { DatetimePickerContainer, DatetimePickerLabel, ChartContainer, ChartLabel, LoaderSpan }