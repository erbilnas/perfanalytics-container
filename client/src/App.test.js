import 'regenerator-runtime'
import { configure, mount, shallow } from 'enzyme'
import 'jest-styled-components'
import Adapter from '@wojtekmaj/enzyme-adapter-react-17'
import React from 'react'
import renderer from 'react-test-renderer'
import { getMeasures } from './services'
import { DatetimePickerContainer, DatetimePickerLabel, ChartContainer, ChartLabel, LoaderSpan } from './styles'
import DatetimePicker from './components/datetimepicker'
import Chart from './components/chart'
import Loader from './components/loader'
import NoData from './components/nodata'

configure({ adapter: new Adapter() });

describe('Get measures from API', () => {
  test('It should get response from /measures API', async () => {
    const [startDate, endDate] = [1, 2]

    return await getMeasures(startDate, endDate).then((response) => expect(response))
  })
})

describe('Get styles', () => {
  test('It should have style rules like position, margin, etc.', () => {
    const containers = shallow(
      <DatetimePickerContainer />,
      <ChartContainer />,
    )

    const datetimePickerLabel = shallow(
      <DatetimePickerLabel />,
    )

    const chartLabel = shallow(
      <ChartLabel />
    )

    const loaderSpan = shallow(
      <LoaderSpan />
    )

    expect(containers).toHaveStyleRule('position', 'absolute')
    expect(datetimePickerLabel).toHaveStyleRule('margin', '0 0 10px 10px')
    expect(chartLabel).toHaveStyleRule('display', 'flex')
    expect(loaderSpan).toHaveStyleRule('margin', '0 0 0 15px')
  })
})

describe('Component: DatetimePicker', () => {
  test('It should render DatetimePicker component', () => {
    const props = { disabledDate: jest.fn(), handler: jest.fn(), showTime: true, datetimeFormat: "string", label: "label" }
    const component = (
      <DatetimePicker
        showTime={props.showTime}
        datetimeFormat={props.datetimeFormat}
        disabledDate={props.disabledDate}
        handler={props.handler}
        label={props.label}
      />
    )

    expect(renderer.create(component).toJSON()).toMatchSnapshot()
  })
})

describe('Component: Chart', () => {
  test('It should render Chart component', () => {
    const props = { data: { a: 1, b: 2 } }
    const component = mount(
      <Chart {...props} />
    )

    expect((component).prop('data')).toEqual({ a: 1, b: 2 })
  })
})

describe('Component: Loader', () => {
  test('It should render Loader component', () => {
    const props = { size: "large" }
    const component = mount(
      <Loader {...props} />
    )

    expect((component).prop('size')).toEqual("large")
  })
})

describe('Component: NoData', () => {
  test('It should render NoData component', () => {
    const props = { description: "desc" }
    const component = mount(
      <NoData {...props} />
    )

    expect((component).prop('description')).toEqual("desc")
  })
})