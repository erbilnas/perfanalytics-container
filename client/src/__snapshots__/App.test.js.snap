// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: Chart It should render DatetimePicker component 1`] = `
Array [
  .c0 {
  margin: 0 0 10px 10px;
}

<p
    className="c0"
  >
    label
  </p>,
  <div
    className="ant-picker ant-picker-range"
    onClick={[Function]}
    onMouseDown={[Function]}
  >
    <div
      className="ant-picker-input ant-picker-input-active"
    >
      <input
        autoComplete="off"
        disabled={false}
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onMouseDown={[Function]}
        placeholder="Start date"
        readOnly={true}
        size={12}
        value=""
      />
    </div>
    <div
      className="ant-picker-range-separator"
    >
      <span
        aria-label="to"
        className="ant-picker-separator"
      >
        <span
          aria-label="swap-right"
          className="anticon anticon-swap-right"
          role="img"
        >
          <svg
            aria-hidden="true"
            data-icon="swap-right"
            fill="currentColor"
            focusable="false"
            height="1em"
            viewBox="0 0 1024 1024"
            width="1em"
          >
            <path
              d="M873.1 596.2l-164-208A32 32 0 00684 376h-64.8c-6.7 0-10.4 7.7-6.3 13l144.3 183H152c-4.4 0-8 3.6-8 8v60c0 4.4 3.6 8 8 8h695.9c26.8 0 41.7-30.8 25.2-51.8z"
            />
          </svg>
        </span>
      </span>
    </div>
    <div
      className="ant-picker-input"
    >
      <input
        autoComplete="off"
        disabled={false}
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onMouseDown={[Function]}
        placeholder="End date"
        readOnly={true}
        size={12}
        value=""
      />
    </div>
    <div
      className="ant-picker-active-bar"
      style={
        Object {
          "left": 0,
          "position": "absolute",
          "width": 0,
        }
      }
    />
    <span
      className="ant-picker-suffix"
    >
      <span
        aria-label="calendar"
        className="anticon anticon-calendar"
        role="img"
      >
        <svg
          aria-hidden="true"
          data-icon="calendar"
          fill="currentColor"
          focusable="false"
          height="1em"
          viewBox="64 64 896 896"
          width="1em"
        >
          <path
            d="M880 184H712v-64c0-4.4-3.6-8-8-8h-56c-4.4 0-8 3.6-8 8v64H384v-64c0-4.4-3.6-8-8-8h-56c-4.4 0-8 3.6-8 8v64H144c-17.7 0-32 14.3-32 32v664c0 17.7 14.3 32 32 32h736c17.7 0 32-14.3 32-32V216c0-17.7-14.3-32-32-32zm-40 656H184V460h656v380zM184 392V256h128v48c0 4.4 3.6 8 8 8h56c4.4 0 8-3.6 8-8v-48h256v48c0 4.4 3.6 8 8 8h56c4.4 0 8-3.6 8-8v-48h128v136H184z"
          />
        </svg>
      </span>
    </span>
  </div>,
]
`;

exports[`Component: DatetimePicker It should get response from /measures API 1`] = `
Array [
  .c0 {
  margin: 0 0 10px 10px;
}

<p
    className="c0"
  >
    label
  </p>,
  <div
    className="ant-picker ant-picker-range"
    onClick={[Function]}
    onMouseDown={[Function]}
  >
    <div
      className="ant-picker-input ant-picker-input-active"
    >
      <input
        autoComplete="off"
        disabled={false}
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onMouseDown={[Function]}
        placeholder="Start date"
        readOnly={true}
        size={12}
        value=""
      />
    </div>
    <div
      className="ant-picker-range-separator"
    >
      <span
        aria-label="to"
        className="ant-picker-separator"
      >
        <span
          aria-label="swap-right"
          className="anticon anticon-swap-right"
          role="img"
        >
          <svg
            aria-hidden="true"
            data-icon="swap-right"
            fill="currentColor"
            focusable="false"
            height="1em"
            viewBox="0 0 1024 1024"
            width="1em"
          >
            <path
              d="M873.1 596.2l-164-208A32 32 0 00684 376h-64.8c-6.7 0-10.4 7.7-6.3 13l144.3 183H152c-4.4 0-8 3.6-8 8v60c0 4.4 3.6 8 8 8h695.9c26.8 0 41.7-30.8 25.2-51.8z"
            />
          </svg>
        </span>
      </span>
    </div>
    <div
      className="ant-picker-input"
    >
      <input
        autoComplete="off"
        disabled={false}
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onMouseDown={[Function]}
        placeholder="End date"
        readOnly={true}
        size={12}
        value=""
      />
    </div>
    <div
      className="ant-picker-active-bar"
      style={
        Object {
          "left": 0,
          "position": "absolute",
          "width": 0,
        }
      }
    />
    <span
      className="ant-picker-suffix"
    >
      <span
        aria-label="calendar"
        className="anticon anticon-calendar"
        role="img"
      >
        <svg
          aria-hidden="true"
          data-icon="calendar"
          fill="currentColor"
          focusable="false"
          height="1em"
          viewBox="64 64 896 896"
          width="1em"
        >
          <path
            d="M880 184H712v-64c0-4.4-3.6-8-8-8h-56c-4.4 0-8 3.6-8 8v64H384v-64c0-4.4-3.6-8-8-8h-56c-4.4 0-8 3.6-8 8v64H144c-17.7 0-32 14.3-32 32v664c0 17.7 14.3 32 32 32h736c17.7 0 32-14.3 32-32V216c0-17.7-14.3-32-32-32zm-40 656H184V460h656v380zM184 392V256h128v48c0 4.4 3.6 8 8 8h56c4.4 0 8-3.6 8-8v-48h256v48c0 4.4 3.6 8 8 8h56c4.4 0 8-3.6 8-8v-48h128v136H184z"
          />
        </svg>
      </span>
    </span>
  </div>,
]
`;

exports[`Component: DatetimePicker It should render DatetimePicker component 1`] = `
Array [
  .c0 {
  margin: 0 0 10px 10px;
}

<p
    className="c0"
  >
    label
  </p>,
  <div
    className="ant-picker ant-picker-range"
    onClick={[Function]}
    onMouseDown={[Function]}
  >
    <div
      className="ant-picker-input ant-picker-input-active"
    >
      <input
        autoComplete="off"
        disabled={false}
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onMouseDown={[Function]}
        placeholder="Start date"
        readOnly={true}
        size={12}
        value=""
      />
    </div>
    <div
      className="ant-picker-range-separator"
    >
      <span
        aria-label="to"
        className="ant-picker-separator"
      >
        <span
          aria-label="swap-right"
          className="anticon anticon-swap-right"
          role="img"
        >
          <svg
            aria-hidden="true"
            data-icon="swap-right"
            fill="currentColor"
            focusable="false"
            height="1em"
            viewBox="0 0 1024 1024"
            width="1em"
          >
            <path
              d="M873.1 596.2l-164-208A32 32 0 00684 376h-64.8c-6.7 0-10.4 7.7-6.3 13l144.3 183H152c-4.4 0-8 3.6-8 8v60c0 4.4 3.6 8 8 8h695.9c26.8 0 41.7-30.8 25.2-51.8z"
            />
          </svg>
        </span>
      </span>
    </div>
    <div
      className="ant-picker-input"
    >
      <input
        autoComplete="off"
        disabled={false}
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onMouseDown={[Function]}
        placeholder="End date"
        readOnly={true}
        size={12}
        value=""
      />
    </div>
    <div
      className="ant-picker-active-bar"
      style={
        Object {
          "left": 0,
          "position": "absolute",
          "width": 0,
        }
      }
    />
    <span
      className="ant-picker-suffix"
    >
      <span
        aria-label="calendar"
        className="anticon anticon-calendar"
        role="img"
      >
        <svg
          aria-hidden="true"
          data-icon="calendar"
          fill="currentColor"
          focusable="false"
          height="1em"
          viewBox="64 64 896 896"
          width="1em"
        >
          <path
            d="M880 184H712v-64c0-4.4-3.6-8-8-8h-56c-4.4 0-8 3.6-8 8v64H384v-64c0-4.4-3.6-8-8-8h-56c-4.4 0-8 3.6-8 8v64H144c-17.7 0-32 14.3-32 32v664c0 17.7 14.3 32 32 32h736c17.7 0 32-14.3 32-32V216c0-17.7-14.3-32-32-32zm-40 656H184V460h656v380zM184 392V256h128v48c0 4.4 3.6 8 8 8h56c4.4 0 8-3.6 8-8v-48h256v48c0 4.4 3.6 8 8 8h56c4.4 0 8-3.6 8-8v-48h128v136H184z"
          />
        </svg>
      </span>
    </span>
  </div>,
]
`;

exports[`Component: DatetimePicker It should render DatetimePicker component 2`] = `
Array [
  .c0 {
  margin: 0 0 10px 10px;
}

<p
    className="c0"
  >
    label
  </p>,
  <div
    className="ant-picker ant-picker-range"
    onClick={[Function]}
    onMouseDown={[Function]}
  >
    <div
      className="ant-picker-input ant-picker-input-active"
    >
      <input
        autoComplete="off"
        disabled={false}
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onMouseDown={[Function]}
        placeholder="Start date"
        readOnly={true}
        size={12}
        value=""
      />
    </div>
    <div
      className="ant-picker-range-separator"
    >
      <span
        aria-label="to"
        className="ant-picker-separator"
      >
        <span
          aria-label="swap-right"
          className="anticon anticon-swap-right"
          role="img"
        >
          <svg
            aria-hidden="true"
            data-icon="swap-right"
            fill="currentColor"
            focusable="false"
            height="1em"
            viewBox="0 0 1024 1024"
            width="1em"
          >
            <path
              d="M873.1 596.2l-164-208A32 32 0 00684 376h-64.8c-6.7 0-10.4 7.7-6.3 13l144.3 183H152c-4.4 0-8 3.6-8 8v60c0 4.4 3.6 8 8 8h695.9c26.8 0 41.7-30.8 25.2-51.8z"
            />
          </svg>
        </span>
      </span>
    </div>
    <div
      className="ant-picker-input"
    >
      <input
        autoComplete="off"
        disabled={false}
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onMouseDown={[Function]}
        placeholder="End date"
        readOnly={true}
        size={12}
        value=""
      />
    </div>
    <div
      className="ant-picker-active-bar"
      style={
        Object {
          "left": 0,
          "position": "absolute",
          "width": 0,
        }
      }
    />
    <span
      className="ant-picker-suffix"
    >
      <span
        aria-label="calendar"
        className="anticon anticon-calendar"
        role="img"
      >
        <svg
          aria-hidden="true"
          data-icon="calendar"
          fill="currentColor"
          focusable="false"
          height="1em"
          viewBox="64 64 896 896"
          width="1em"
        >
          <path
            d="M880 184H712v-64c0-4.4-3.6-8-8-8h-56c-4.4 0-8 3.6-8 8v64H384v-64c0-4.4-3.6-8-8-8h-56c-4.4 0-8 3.6-8 8v64H144c-17.7 0-32 14.3-32 32v664c0 17.7 14.3 32 32 32h736c17.7 0 32-14.3 32-32V216c0-17.7-14.3-32-32-32zm-40 656H184V460h656v380zM184 392V256h128v48c0 4.4 3.6 8 8 8h56c4.4 0 8-3.6 8-8v-48h256v48c0 4.4 3.6 8 8 8h56c4.4 0 8-3.6 8-8v-48h128v136H184z"
          />
        </svg>
      </span>
    </span>
  </div>,
]
`;

exports[`Component: Loader It should render DatetimePicker component 1`] = `
Array [
  .c0 {
  margin: 0 0 10px 10px;
}

<p
    className="c0"
  >
    label
  </p>,
  <div
    className="ant-picker ant-picker-range"
    onClick={[Function]}
    onMouseDown={[Function]}
  >
    <div
      className="ant-picker-input ant-picker-input-active"
    >
      <input
        autoComplete="off"
        disabled={false}
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onMouseDown={[Function]}
        placeholder="Start date"
        readOnly={true}
        size={12}
        value=""
      />
    </div>
    <div
      className="ant-picker-range-separator"
    >
      <span
        aria-label="to"
        className="ant-picker-separator"
      >
        <span
          aria-label="swap-right"
          className="anticon anticon-swap-right"
          role="img"
        >
          <svg
            aria-hidden="true"
            data-icon="swap-right"
            fill="currentColor"
            focusable="false"
            height="1em"
            viewBox="0 0 1024 1024"
            width="1em"
          >
            <path
              d="M873.1 596.2l-164-208A32 32 0 00684 376h-64.8c-6.7 0-10.4 7.7-6.3 13l144.3 183H152c-4.4 0-8 3.6-8 8v60c0 4.4 3.6 8 8 8h695.9c26.8 0 41.7-30.8 25.2-51.8z"
            />
          </svg>
        </span>
      </span>
    </div>
    <div
      className="ant-picker-input"
    >
      <input
        autoComplete="off"
        disabled={false}
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onMouseDown={[Function]}
        placeholder="End date"
        readOnly={true}
        size={12}
        value=""
      />
    </div>
    <div
      className="ant-picker-active-bar"
      style={
        Object {
          "left": 0,
          "position": "absolute",
          "width": 0,
        }
      }
    />
    <span
      className="ant-picker-suffix"
    >
      <span
        aria-label="calendar"
        className="anticon anticon-calendar"
        role="img"
      >
        <svg
          aria-hidden="true"
          data-icon="calendar"
          fill="currentColor"
          focusable="false"
          height="1em"
          viewBox="64 64 896 896"
          width="1em"
        >
          <path
            d="M880 184H712v-64c0-4.4-3.6-8-8-8h-56c-4.4 0-8 3.6-8 8v64H384v-64c0-4.4-3.6-8-8-8h-56c-4.4 0-8 3.6-8 8v64H144c-17.7 0-32 14.3-32 32v664c0 17.7 14.3 32 32 32h736c17.7 0 32-14.3 32-32V216c0-17.7-14.3-32-32-32zm-40 656H184V460h656v380zM184 392V256h128v48c0 4.4 3.6 8 8 8h56c4.4 0 8-3.6 8-8v-48h256v48c0 4.4 3.6 8 8 8h56c4.4 0 8-3.6 8-8v-48h128v136H184z"
          />
        </svg>
      </span>
    </span>
  </div>,
]
`;

exports[`Component: NoData It should render DatetimePicker component 1`] = `
Array [
  .c0 {
  margin: 0 0 10px 10px;
}

<p
    className="c0"
  >
    label
  </p>,
  <div
    className="ant-picker ant-picker-range"
    onClick={[Function]}
    onMouseDown={[Function]}
  >
    <div
      className="ant-picker-input ant-picker-input-active"
    >
      <input
        autoComplete="off"
        disabled={false}
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onMouseDown={[Function]}
        placeholder="Start date"
        readOnly={true}
        size={12}
        value=""
      />
    </div>
    <div
      className="ant-picker-range-separator"
    >
      <span
        aria-label="to"
        className="ant-picker-separator"
      >
        <span
          aria-label="swap-right"
          className="anticon anticon-swap-right"
          role="img"
        >
          <svg
            aria-hidden="true"
            data-icon="swap-right"
            fill="currentColor"
            focusable="false"
            height="1em"
            viewBox="0 0 1024 1024"
            width="1em"
          >
            <path
              d="M873.1 596.2l-164-208A32 32 0 00684 376h-64.8c-6.7 0-10.4 7.7-6.3 13l144.3 183H152c-4.4 0-8 3.6-8 8v60c0 4.4 3.6 8 8 8h695.9c26.8 0 41.7-30.8 25.2-51.8z"
            />
          </svg>
        </span>
      </span>
    </div>
    <div
      className="ant-picker-input"
    >
      <input
        autoComplete="off"
        disabled={false}
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onMouseDown={[Function]}
        placeholder="End date"
        readOnly={true}
        size={12}
        value=""
      />
    </div>
    <div
      className="ant-picker-active-bar"
      style={
        Object {
          "left": 0,
          "position": "absolute",
          "width": 0,
        }
      }
    />
    <span
      className="ant-picker-suffix"
    >
      <span
        aria-label="calendar"
        className="anticon anticon-calendar"
        role="img"
      >
        <svg
          aria-hidden="true"
          data-icon="calendar"
          fill="currentColor"
          focusable="false"
          height="1em"
          viewBox="64 64 896 896"
          width="1em"
        >
          <path
            d="M880 184H712v-64c0-4.4-3.6-8-8-8h-56c-4.4 0-8 3.6-8 8v64H384v-64c0-4.4-3.6-8-8-8h-56c-4.4 0-8 3.6-8 8v64H144c-17.7 0-32 14.3-32 32v664c0 17.7 14.3 32 32 32h736c17.7 0 32-14.3 32-32V216c0-17.7-14.3-32-32-32zm-40 656H184V460h656v380zM184 392V256h128v48c0 4.4 3.6 8 8 8h56c4.4 0 8-3.6 8-8v-48h256v48c0 4.4 3.6 8 8 8h56c4.4 0 8-3.6 8-8v-48h128v136H184z"
          />
        </svg>
      </span>
    </span>
  </div>,
]
`;

exports[`Get styles It should get response from /measures API 1`] = `ReactWrapper {}`;
